lib/common/results.c

```diff
@@ -852,6 +853,51 @@ pcmk_rc2ocf(int rc)
 
 // Other functions
 
+/*!
+ * \brief Map a getaddrinfo() return code to the most similar Pacemaker
+ *        return code
+ *
+ * \param[in] gai  getaddrinfo() return code
+ *
+ * \return Most similar Pacemaker return code
+ */
+int
+pcmk_gaierror2rc(int gai)
+{
+    switch (gai) {
+        case 0:
+            return pcmk_rc_ok;
+
+        case EAI_ADDRFAMILY:
+        case EAI_FAIL:
+        case EAI_NODATA:
+        case EAI_NONAME:
+            return pcmk_rc_ns_resolution;
+
+        case EAI_AGAIN:
+            return EAGAIN;
+
+        case EAI_BADFLAGS:
+        case EAI_SERVICE:
+            return pcmk_rc_net_software;
+
+        case EAI_FAMILY:
+            return EAFNOSUPPORT;
+
+        case EAI_MEMORY:
+            return ENOMEM;
+
+        case EAI_SOCKTYPE:
+            return ESOCKTNOSUPPORT;
```

Comment:
ESOCKTNOSUPPORT isn't POSIX, but as long as it compiles on all CI platforms we can ignore that (otherwise, we can define it in portability.h the same way we do ENOTUNIQ etc.)

Reply:
I'll double check the CI results next time I push changes.

Reply:
The only CI systems that failed were the rawhide ones, and that's only because of the python tests not running.